npm WARN deprecated @hapi/joi@14.5.0: This version is no longer maintained

<!--
  Please fill out each section below, otherwise your issue will be closed. This info allows Gatsby maintainers to diagnose (and fix!) your issue as quickly as possible.

  Useful Links:
  - Documentation: https://www.gatsbyjs.org/docs/
  - How to File an Issue: https://www.gatsbyjs.org/contributing/how-to-file-an-issue/

  Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
-->

## Description

Describe the issue that you're seeing. When I updated to gatsby 2.11.3, as version 2.11.3 was being downloaded, I got the following warning in Terminal:

`npm WARN deprecated @hapi/joi@14.5.0: This version is no longer maintained`

 Subsequently, I got the following error in Terminal when running npx gatsby develop:

```
Error in "/Users/mariacam/Development/interglobalmedia/node_modules/gatsby-plugin-feed/gatsby-node.js": Cannot find module 'joi'



  Error: Cannot find module 'joi'

  - loader.js:668 Function.Module._resolveFilename
    internal/modules/cjs/loader.js:668:15

  - loader.js:591 Function.Module._load
    internal/modules/cjs/loader.js:591:27

  - loader.js:723 Module.require
    internal/modules/cjs/loader.js:723:19

  - v8-compile-cache.js:159 require
    [interglobalmedia]/[v8-compile-cache]/v8-compile-cache.js:159:20

  - plugin-options.js:8 Object.<anonymous>
    [interglobalmedia]/[gatsby-plugin-feed]/plugin-options.js:8:35

  - v8-compile-cache.js:178 Module._compile
    [interglobalmedia]/[v8-compile-cache]/v8-compile-cache.js:178:30

  - loader.js:827 Object.Module._extensions..js
    internal/modules/cjs/loader.js:827:10

  - loader.js:685 Module.load
    internal/modules/cjs/loader.js:685:32

  - loader.js:620 Function.Module._load
    internal/modules/cjs/loader.js:620:12

  - loader.js:723 Module.require
    internal/modules/cjs/loader.js:723:19

  - v8-compile-cache.js:159 require
    [interglobalmedia]/[v8-compile-cache]/v8-compile-cache.js:159:20

  - gatsby-node.js:23 Object.<anonymous>
    [interglobalmedia]/[gatsby-plugin-feed]/gatsby-node.js:23:45

  - v8-compile-cache.js:178 Module._compile
    [interglobalmedia]/[v8-compile-cache]/v8-compile-cache.js:178:30

  - loader.js:827 Object.Module._extensions..js
    internal/modules/cjs/loader.js:827:10

  - loader.js:685 Module.load
    internal/modules/cjs/loader.js:685:32

  - loader.js:620 Function.Module._load
    internal/modules/cjs/loader.js:620:12


⠋ load plugins

```
### Steps to reproduce 

`npm i gatsby@latest -S`

`npx gatsby develop`

or

`npm run build`

Clear steps describing how to reproduce the issue. Please please please link to a demo project if possible, this makes your issue _much_ easier to diagnose (seriously).

link to repo (does not have 2.11.3 for obvious reasons, but could be added. Everything else is the same): [interglobalmedia repo](https://github.com/interglobalmedia/interglobalmedia)

### Expected result

with nxp gatsby develop, that my localhost runs and that my project renders in the browser without crashing.

What should happen? The development server should run and the project render in the browser.

### Actual result My project crashed before the development server could even start.

What happened.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```
System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.14.0 - ~/.nvm/versions/node/v11.14.0/bin/node
    npm: 6.7.0 - ~/.nvm/versions/node/v11.14.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 75.0.3770.100
  npmPackages:
    gatsby: ^2.11.3 => 2.11.3
    gatsby-image: ^2.1.4 => 2.1.4
    gatsby-plugin-feed: ^2.3.0 => 2.3.0
    gatsby-plugin-google-analytics: ^2.0.21 => 2.0.21
    gatsby-plugin-gtag: ^1.0.11 => 1.0.11
    gatsby-plugin-manifest: ^2.1.1 => 2.1.1
    gatsby-plugin-offline: ^2.1.3 => 2.1.3
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-sass: ^2.0.11 => 2.0.11
    gatsby-plugin-sharp: ^2.1.5 => 2.1.5
    gatsby-plugin-sitemap: ^2.1.0 => 2.1.0
    gatsby-plugin-styled-components: ^3.0.7 => 3.0.7
    gatsby-plugin-twitter: ^2.0.13 => 2.0.13
    gatsby-plugin-typography: ^2.2.13 => 2.2.13
    gatsby-remark-copy-linked-files: ^2.0.13 => 2.0.13
    gatsby-remark-images: ^3.0.16 => 3.0.16
    gatsby-remark-prismjs: ^3.2.11 => 3.2.11
    gatsby-remark-responsive-iframe: ^2.1.1 => 2.1.1
    gatsby-remark-smartypants: ^2.0.9 => 2.0.9
    gatsby-source-filesystem: ^2.0.39 => 2.0.39
    gatsby-transformer-remark: ^2.4.0 => 2.4.0
    gatsby-transformer-sharp: ^2.1.21 => 2.1.21

```

--- a/packages/gatsby-plugin-feed/package.json
+++ b/packages/gatsby-plugin-feed/package.json
@@ -8,7 +8,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "@hapi/joi": "^14.0.0",
+    "@hapi/joi": "^15.0.0",
     "fs-extra": "^7.0.1",
     "lodash.merge": "^4.6.0",
     "rss": "^1.2.2"

--- a/packages/gatsby-source-contentful/package.json
+++ b/packages/gatsby-source-contentful/package.json
@@ -8,7 +8,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "@hapi/joi": "^14.0.0",
+    "@hapi/joi": "^15.0.0",
     "axios": "^0.19.0",
     "base64-img": "^1.0.3",
     "bluebird": "^3.5.0",

--- a/packages/gatsby/package.json
+++ b/packages/gatsby/package.json
@@ -17,7 +17,7 @@
     "@babel/runtime": "^7.0.0",
     "@babel/traverse": "^7.0.0",
     "@gatsbyjs/relay-compiler": "2.0.0-printer-fix.2",
-    "@hapi/joi": "^14.0.0",
+    "@hapi/joi": "^15.0.0",
     "@mikaelkristiansson/domready": "^1.0.9",
     "@pieh/friendly-errors-webpack-plugin": "1.7.0-chalk-2",
     "@reach/router": "^1.1.1",

